File: returned/rotate-array.ts
/**
 * https://leetcode.com/problems/rotate-array/?envType=study-plan-v2&envId=top-interview-150
 * Given an integer array nums, rotate the array to the right by k steps, where k is non-negative.
 */

export { rotate };

function rotate(nums: number[], k: number): void {
  const result: number[] = [];
  let i = nums.length - (k % nums.length);
  while (result.length < nums.length) {
    if (i >= nums.length) i = 0;
    result.push(nums[i]);
    i++;
  }
  for (let i = 0; i < result.length; i++) {
    nums[i] = result[i];
  }
}

File: csharp/Leet/Old/RotateArray.cs
// https://leetcode.com/problems/rotate-array/description/
namespace Leet.RotateArray
{
    public class Solution
    {
        public void Rotate(int[] nums, int k)
        {
            k = k % nums.Length;
            var left = nums[..^k];
            var right = nums[(nums.Length - k)..];
            var result = right.Concat(left).ToArray();

            for (var i = 0; i < nums.Length; i++)
            {
                nums[i] = result[i];
            }
        }
    }
}
